src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+            clobber=clobber,
+        )
+
+    def _create_date_list(
+        self, start_year, end_year, start_month, end_month, start_day, end_day
+    ):
+
+        # Check whether the input dates are valid, assign dates when not
+        # specified and create list of tuples containing the range of
+        # dates of interest
+
+        if (
+            start_month is not None or start_day is not None
+        ) and start_year is None:
+            raise ValueError(
+                "If you specify starting month or day, you also "
+                "need to specify the starting year."
+            )
+
+        if (end_month is not None or end_day is not None) and end_year is None:
+            raise ValueError(
+                "If you specify end month or day, you also "
+                "need to specify the end year."
+            )
+
+        if start_day is not None and start_month is None:
+            raise ValueError(
+                "If you specify the starting day, you also "
+                "need to specify the starting month."
+            )
+
+        if end_day is not None and end_month is None:
+            raise ValueError(
+                "If you specify the end day, you also "
+                "need to specify the end month."
+            )
+
+        start_avail_date = date(year=1981, month=1, day=1)
+        end_avail_date = self._get_last_available_date()
+
+        if start_year is None:
+            start_year = start_avail_date.year
+            start_month = start_avail_date.month
+            start_day = start_avail_date.day
+        if end_year is None:
+            end_year = end_avail_date.year
+            end_month = end_avail_date.month
+            end_day = end_avail_date.day
+
+        if start_month is None:
+            start_month = 1
+        if end_month is None:
+            if end_year == end_avail_date.year:
+                end_month = end_avail_date.month
+            else:
+                end_month = 12
+
+        if start_day is None:
+            start_day = 1
+        if end_day is None:
+            if (
+                end_year == end_avail_date.year
+                and end_month == end_avail_date.month
+            ):
+                end_day = end_avail_date.day
+            else:
+                end_day = calendar.monthrange(end_year, end_month)[1]
+
+        try:
+            start_date = date(
+                year=start_year, month=start_month, day=start_day
+            )
+            end_date = date(year=end_year, month=end_month, day=end_day)
+        except ValueError:
+            raise ValueError(
+                "(One of) the dates identified by the "
+                "given inputs are not valid."
+            )
+
+        if not start_avail_date <= start_date <= end_date <= end_avail_date:
+            raise ValueError(
+                "Check that the input dates are ordered in the "
+                f"following way: {start_avail_date} <= start date "
```

Comment:
I would also use the `start_date` and `end_date` vars in this string instead of only showing "start date"